docs/gallery/plot_term_set.py

```diff
@@ -0,0 +1,146 @@
+"""
+TermSet
+=======
+
+This is a user guide for interacting with the
+:py:class:`~hdmf.TermSet` class. The :py:class:`~hdmf.TermSet` type
+is experimental and is subject to change in future releases. If you use this type,
+please provide feedback to the HDMF team so that we can improve the structure and
+overall capabilities.
+
+Introduction
+-------------
+The :py:class:`~hdmf.TermSet` class provides a way for users to create their own
+set of terms from brain atlases, species taxonomies, and anatomical, cell, and
+gene function ontologies.
+
+:py:class:`~hdmf.TermSet` serves two purposes: data validation and external reference
+management. Users will be able to validate their data to their own set of terms, ensuring
+clean data to be used inline with the FAIR principles later on.
+The  :py:class:`~hdmf.TermSet` class allows for a reusable and sharable
+pool of metadata to serve as references to any dataset within the NWB ecosystem.
```

Comment:
```suggestion
pool of metadata to serve as references to any dataset.
```

This is a general feature - not just for NWB (though that is our primary use case).